include/aws_iot_mqtt_client.h

```diff
@@ -312,112 +312,182 @@ struct _Client {
+ *
+ * The new disconnect handler takes effect when the next disconnect is detected.
  *
- * Called to set the IoT Client disconnect handler
- * The disconnect handler is called whenever the client disconnects with error
+ * @param[in] pClient MQTT client context
+ * @param[in] pDisconnectHandler New disconnect handler
+ * @param[in] pDisconnectHandlerData Context to be passed to new disconnect handler
  *
- * @param pClient Reference to the IoT Client
- * @param pDisconnectHandler Reference to the new Disconnect Handler
- * @param pDisconnectHandlerData Reference to the data to be passed as argument when disconnect handler is called
+ * @return Returns NULL_VALUE_ERROR if provided a bad parameter; otherwise, always
+ * returns SUCCESS.
  *
- * @return IoT_Error_t Type defining successful/failed API call
+ * @warning Do not call this function if @ref mqtt_function_yield is in progress.
  */
+/* @[declare_mqtt_set_disconnect_handler] */
 IoT_Error_t aws_iot_mqtt_set_disconnect_handler(AWS_IoT_Client *pClient, iot_disconnect_handler pDisconnectHandler,
 												void *pDisconnectHandlerData);
+/* @[declare_mqtt_set_disconnect_handler] */
 
 /**
- * @brief Enable or Disable AutoReconnect on Network Disconnect
+ * @brief Enable or disable auto-reconnect for an initialized MQTT client context.
  *
- * Called to enable or disabled the auto reconnect features provided with the SDK
+ * This function replaces the current auto-reconnect setting with the provided setting.
  *
- * @param pClient Reference to the IoT Client
- * @param newStatus set to true for enabling and false for disabling
+ * @note This function should only be called after @ref mqtt_function_connect has been
+ * called for the provided client.
  *
- * @return IoT_Error_t Type defining successful/failed API call
+ * @param[in] pClient MQTT client context
+ * @param[in] newStatus New setting for auto-reconnect
+ *
+ * @return Returns NULL_VALUE_ERROR if provided a bad parameter; otherwise, always
+ * returns SUCCESS.
+ *
+  * @warning Do not call this function if a connection attempt is in progress. Connection
+ * attempts happen in the context of @ref mqtt_function_connect, @ref mqtt_function_attempt_reconnect,
+ * or @ref mqtt_function_yield.
  */
+/* @[declare_mqtt_autoreconnect_set_status] */
 IoT_Error_t aws_iot_mqtt_autoreconnect_set_status(AWS_IoT_Client *pClient, bool newStatus);
+/* @[declare_mqtt_autoreconnect_set_status] */
 
 /**
- * @brief Get count of Network Disconnects
+ * @brief Get the current number of disconnects detected by an MQTT client context.
+ *
+ * @param[in] pClient MQTT client context
  *
- * Called to get the number of times a network disconnect occurred due to errors
+ * @return The number of disconnects detected since the client was created
+ * (or since the last call to @ref mqtt_function_reset_network_disconnected_count).
  *
- * @param pClient Reference to the IoT Client
+ * @warning Application code should not rely on this function's return value.
```

Comment:
Is it only true for a multi-threaded environment?

Reply:
Correct.  Removed line.